examples/opencv_rc.py

```diff
@@ -0,0 +1,541 @@
+        self._stop_all()
+
+        # Bring the lift down
+        self.lift_height = pc.MIN_LIFT_HEIGHT.mm
+
+        # Set the head down as well
+        self.head_tilt = pc.MIN_HEAD_ANGLE.radians
+
+        # Turn off the light
+        self.head_light = False
+
+        # Disconnect from Cozmo
+        self._cozmo_clt.disconnect()
+        self._cozmo_clt.stop()
+
+        # If the keyboard listener is still running
+        if self._kbd_listener.running:
+            self._kbd_listener.stop()
+            self._kbd_listener.join()
+
+        # Close any display open by OpenCv
+        cv.destroyAllWindows()
+
+    def _on_new_image(self, cli, frame):
+        """
+        A simple function that converts a frame from Cozmo's camera into a
+        up-scaled and BGR formatted image to be used by OpenCV.
+        :param cli: An instance of the pycozmo. Client class representing the
+        robot.
+        :param frame: A color/grayscale frame from Cozmo's camera.
+        :return: None
+        """
+
+        # Convert the image into a numpy array so that OpenCV can manipulate it
+        orig_img = np.array(frame)
+
+        # Check if we got a color image
+        if orig_img.shape[-1] == 3:
+            # The thing about OpenCV is that it uses BGR formatted images for
+            # reasons
+            orig_img = cv.cvtColor(orig_img, cv.COLOR_RGB2BGR)
+
+        # Resize the image
+        # The lanczos4 algorithm produces the best results, but might be slow
+        # you can use cv.INTER_LINEAR for poorer, but faster results
+        resized_img = cv.resize(orig_img, None, fx=2, fy=2,
+                                interpolation=cv.INTER_LANCZOS4)
+
+        # Try to reduce the noise using unsharp masking
+        # An explanation for this technique can be found here:
+        # https://en.wikipedia.org/wiki/Unsharp_masking#Digital_unsharp_masking
+        blurred_img = cv.GaussianBlur(resized_img, (3, 3), 0)
+        sharp_img = cv.addWeighted(resized_img, 1 + self._sharp_amount,
+                                   blurred_img, -self._sharp_amount,
+                                   gamma=self._sharp_gamma)
+
+        # NOTE: This could be used with cv.filter2D() in place of the unsharp
+        # masking. However, controlling the amount of sharpening is more
+        # difficult
+        # UNSHARP_KERNEL = -1 / 256 * np.array([[1, 4, 6, 4, 1],
+        #                                       [4, 16, 24, 16, 4],
+        #                                       [6, 24, -476, 24, 6],
+        #                                       [4, 16, 24, 16, 4],
+        #                                       [1, 4, 6, 4, 1]])
+
+        # Display the image in the video feed window
+        cv.imshow(self._win_name, sharp_img)
+        # This might seem odd, but is actually required by OpenCV to perform GUI
+        # housekeeping. See OpenCV's documentation for imshow() for more
+        # "in-depth" information.
+        cv.waitKey(25)
```

Comment:
`_on_new_image` is being called whenever a new frame is available. The frames are streamed as recorded, which means that you don't need to wait here. I don't think the problem is here, but the lagging issues you are seeing could be related to this.

Reply:
As I mention in the comment on line 205-207, this wait is actually required by OpenCV. As per the [documentation](https://docs.opencv.org/master/d7/dfc/group__highgui.html#ga453d42fe4cb60e5723281a89973ee563) a call to `cv.imshow()` should always be followed by a call to `cv.waitKey()`, otherwise the image will not be displayed. The best I can do here is to set the wait to only 1 ms. But as you suggested in your overall comment, I will try putting the actual display function back into the main thread and have it communicate via queue with the callback.

Reply:
Yes you are right, it might be causing trouble here because of the callback though. I think this should go separately from the frame retrieval